src/components/remote_control/test/commands/on_remote_control_settings_test.cc

```diff
@@ -0,0 +1,173 @@
+using remote_control::MessageHelper;
+using namespace remote_control;
+
+namespace {
+const int kModuleId = 153;
+const std::string kAllowedTrueJson =
+    "{ \"jsonrpc\": \"2.0\",\"method\": \"RC.OnRemoteControlSettings\",\
+         \"params\":{\"accessMode\": \"AUTO_ALLOW\", \"allowed\":true}}";
+const std::string kAllowedFalseJson =
+    "{ \"jsonrpc\": \"2.0\",\"method\": \"RC.OnRemoteControlSettings\",\
+         \"params\":{\"accessMode\": \"AUTO_ALLOW\", \"allowed\":false}}";
+}
+
+namespace test {
+namespace components {
+namespace remote_control_test {
+namespace on_remote_control_settings_notification_test {
+
+class OnRemoteControlSettingsNotificationTest : public ::testing::Test {
+ public:
+  OnRemoteControlSettingsNotificationTest()
+      : mock_service_(utils::MakeShared<NiceMock<MockService> >())
+      , mock_app_(utils::MakeShared<NiceMock<MockApplication> >())
+      , rc_app_extention_(
+            utils::MakeShared<remote_control::RCAppExtension>(kModuleId)) {
+    ON_CALL(mock_module_, service()).WillByDefault(Return(mock_service_));
+    ON_CALL(mock_module_, resource_allocation_manager())
+        .WillByDefault(ReturnRef(mock_allocation_manager_));
+    ServicePtr exp_service(mock_service_);
+    mock_module_.set_service(exp_service);
+  }
+
+  remote_control::request_controller::MobileRequestPtr CreateCommand(
+      application_manager::MessagePtr msg) {
+    return remote_control::RCCommandFactory::CreateCommand(msg, mock_module_);
+  }
+
+  application_manager::MessagePtr CreateBasicMessage() {
+    application_manager::MessagePtr message = utils::MakeShared<Message>(
+        MessagePriority::FromServiceType(protocol_handler::ServiceType::kRpc));
+    message->set_function_id(RCFunctionID::ON_REMOTE_CONTROL_SETTINGS);
+    message->set_function_name("OnRemoteControlSettings");
+    return message;
+  }
+
+ protected:
+  utils::SharedPtr<NiceMock<application_manager::MockService> > mock_service_;
+  utils::SharedPtr<NiceMock<MockApplication> > mock_app_;
+  utils::SharedPtr<remote_control::RCAppExtension> rc_app_extention_;
+  remote_control_test::MockRemotePluginInterface mock_module_;
+  std::vector<ApplicationSharedPtr> apps_;
+  testing::NiceMock<remote_control_test::MockResourceAllocationManager>
+      mock_allocation_manager_;
+};
+
+TEST_F(OnRemoteControlSettingsNotificationTest,
+       Execute_SetAccessMode_IfAllowedTrue) {
+  // Arrange
+  application_manager::MessagePtr message = CreateBasicMessage();
+  message->set_json_message(kAllowedTrueJson);
+  apps_.push_back(mock_app_);
+  const Json::Value value = MessageHelper::StringToValue(kAllowedTrueJson);
+  const std::string access_mode =
+      value[json_keys::kParams][message_params::kAccessMode].asString();
+  const hmi_apis::Common_RCAccessMode::eType access_mode_ =
+      MessageHelper::AccessModeFromString(access_mode);
+  // Expectations
+  EXPECT_CALL(mock_module_, service()).WillOnce(Return(mock_service_));
```

Comment:
@LitvinenkoIra you have already `service()` call expectation in test class ctor. Either remove it from test fixture or remove expectation from ctor.